apps/docs/docs/state/setup.mdx

```diff
@@ -0,0 +1,645 @@
+---
+sidebar_label: Setup
+sidebar_position: 2
+title: Setup
+# Renamed from libs/state/docs/usage.md
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Basic Setup
+
+The new functional creation API lets you create and configure `RxState` in only one place.
+
+:::info Migration Guide
+
+Read the [following section](#migrate-to-new-functional-api) for a migration guide explaining how to upgrade your codebase to the new API.
+
+:::
+
+```ts
+import { rxState } from '@rx-angular/state';
+import { RxFor } from '@rx-angular/template/for';
+
+@Component({
+  template: `<movie *rxFor="let movie of movies$" [movie]="movie" />`,
+  imports: [RxFor],
+})
+export class MovieListComponent {
+  private state = rxState<{ movies: Movie[] }>(({ set }) => {
+    // set initial state
+    set({ movies: [] });
+  });
+
+  // select a property for the template to consume
+  movies$ = this.state.select('movies');
+}
+```
+
+The functional approach will be the new default approach for newer versions.
+
+Read the [Migration Guide](https://rx-angular.io/docs/state/setup#migrate-to-new-functional-api) for a migration guide explaining how to upgrade your codebase to the new API.
+
+## Connect global state
+
+**Connect state slices from third-party services (e.g. NgRx `Store`)**
+
+Many people have problems combining observables with the component state in a clean way.
+Here is a use case where the `@ngrx/store` gets connected to the local state:
+
+```typescript
+import { rxState } from '@rx-angular/state';
+import { RxFor } from '@rx-angular/template/for';
+
+@Component({
+  template: `<movie *rxFor="let movie of movies$" [movie]="movie" />`,
+  imports: [RxFor],
+})
+export class MovieListComponent {
+  private store = inject(Store<MovieState>);
+
+  private state = rxState<{ movies: Movie[] }>(({ set, connect }) => {
+    // set initial state
+    set({ movies: [] });
+    // connect global state to your local state
+    connect('movies', this.store.select('movies'));
+  });
+
+  // select a property for the template to consume
+  movies$ = this.state.select('movies');
+}
+```
+
+## Store loading & error information
+
+`RxState`s API makes it extremely easy to derive and store loading & error information when interacting with 3rd party data.
+Using one of the overloads of the `connect` method, we can fill our whole state object with only one connection.
+
+```typescript
+import { Component, inject } from '@angular/core';
+import { rxState } from '@rx-angular/state';
+import { RxFor } from '@rx-angular/template/for';
+import { RxIf } from '@rx-angular/template/if';
+import { map, catchError, startWith, endWith } from 'rxjs';
+
+@Component({
+  template: `
+    <loader *rxIf="loading$" />
+    <error *rxIf="error$" />
+    <movie *rxFor="let movie of movies$" [movie]="movie" />
+  `,
+  imports: [RxFor, RxIf],
```

Comment:
Shouldn't we add loader, error and movie components in the imports for the example to work correctly when copied?

Reply:
Then we should add all missing properties like standalone, selector, imports... I would suggest keeping the example minimal.